notebooks/WFC3/point_spread_function/psf_utilities.py

```
@@ -0,0 +1,402 @@
+    The size of the resulting cutout is set by 'rpix' and the red markers show the center value of 'rpix'.
+    '''
+    
+    figure, mysubplot = matplotlib.pyplot.subplots(1, 4, figsize=(11, 11), sharex=True, sharey=True)
+    mysubplot[0].imshow(data, vmin=0.0, vmax=numpy.amax(data), origin='lower')
+    mysubplot[1].imshow(data, vmin=0.0, vmax=numpy.amax(data)/100.0, origin='lower')
+    mysubplot[2].imshow(data, norm=LogNorm(vmin=percentile(data, 0.0), vmax=percentile(data, 98.5)), origin='lower', aspect='equal')
+    mysubplot[3].imshow(numpy.log10(data), origin='lower')
+    mysubplot[0].set_title(r'100\% Max')
+    mysubplot[1].set_title(r'1\% Max')
+    mysubplot[2].set_title('LogNorm')
+    mysubplot[3].set_title('Log10')
+    for idx in [0, 1, 2, 3]:
+        mysubplot[idx].scatter(rpix, rpix, c='red', marker='+')
+    matplotlib.pyplot.tight_layout()
+    matplotlib.pyplot.show()
+
+    return figure
+
+
+def stack_cutouts(input_array, rpix, stack_type='median', scale_flux=True, export_file='none'):
+
+    '''
+    Given an array containing 2D arrays with centered images of stars, create and mean or median stack.
+    The 'rpix' parameter sets the size of the resulting array, and a 'median' stack_type is generally 
+    recommended for stars so that outliers and contaminants are rejected. The scale_flux option allows 
+    users to normalize the total integrated flux (area under the 2D curve) to unity, which is typically 
+    required for PSF models that will be used for PSF fitting or PSF matching with convolution kernels.
+    '''
+
+    print('Calling stack_cutouts with', len(input_array), 'sources.')
+
+    if (stack_type == 'mean'):
+        stacked_image = numpy.mean(input_array, axis=0)
+    elif (stack_type == 'median'):
+        stacked_image = numpy.median(input_array, axis=0)
+    else:
+        return print('ERROR: stack_type must be mean or median.')
```

Raise error instead of printing
```
raise ValueError('stack_type must be mean or median.')
```

---

Done.